docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,122 @@
+
+- [Dialog](/material-ui/react-dialog/)
+- [Drawer](/material-ui/react-drawer/)
+- [Menu](/material-ui/react-menu/)
+- [Popover](/material-ui/react-popover/)
+
+## Basic modal
+
+```js
+import ModalUnstyled from '@mui/base/ModalUnstyled';
+```
+
+{{"demo": "ModalUnstyled.js", "defaultCodeOpen": false }}
+
+## Nested modal
+
+Modals can be nested, for example a select within a dialog, but stacking of more than two modals, or any two modals with a backdrop is discouraged.
+
+{{"demo": "NestedModal.js", "defaultCodeOpen": false}}
+
+## Transitions
+
+The open/close state of the modal can be animated with a transition component.
+This component should respect the following conditions:
+
+- Be a direct child descendent of the modal.
+- Have an `in` prop. This corresponds to the open/close state.
+- Call the `onEnter` callback prop when the enter transition starts.
+- Call the `onExited` callback prop when the exit transition is completed.
+  These two callbacks allow the modal to unmount the child content when closed and fully transitioned.
+
+Modal has built-in support for [react-transition-group](https://github.com/reactjs/react-transition-group).
+
+{{"demo": "TransitionsModal.js", "defaultCodeOpen": false}}
+
+Alternatively, you can use [react-spring](https://github.com/pmndrs/react-spring).
+
+{{"demo": "SpringModal.js", "defaultCodeOpen": false}}
+
+## Performance
+
+The content of modal is unmounted when closed.
+If you need to make the content available to search engines or render expensive component trees inside your modal while optimizing for interaction responsiveness
+it might be a good idea to change this default behavior by enabling the `keepMounted` prop:
+
+```jsx
+<Modal keepMounted />
+```
+
+{{"demo": "KeepMountedModal.js", "defaultCodeOpen": false}}
+
+> You can use the `MuiModal-hidden` class to hide the modal when it is not opened.
+
+As with any performance optimization, this is not a silver bullet.
+Be sure to identify bottlenecks first, and then try out these optimization strategies.
+
+## Server-side modal
+
+React [doesn't support](https://github.com/facebook/react/issues/13097) the [`createPortal()`](https://reactjs.org/docs/portals.html) API on the server.
+In order to display the modal, you need to disable the portal feature with the `disablePortal` prop:
+
+{{"demo": "ServerModal.js", "defaultCodeOpen": false}}
+
+## Limitations
+
+### Focus trap
+
+The modal moves the focus back to the body of the component if the focus tries to escape it.
+
+This is done for accessibility purposes. However, it might create issues.
+In the event the users need to interact with another part of the page, e.g. with a chatbot window, you can disable the behavior:
+
+```jsx
+<Modal disableEnforceFocus />
+```
+
+## Accessibility
+
+(WAI-ARIA: https://www.w3.org/TR/wai-aria-practices/#dialog_modal)
+
+- Be sure to add `aria-labelledby="id..."`, referencing the modal title, to the `Modal`.
```

Comment:
Let's explain _why_ the developer should do this.

```suggestion
- All interactive elements must have an [accessible name](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby). Use the `aria-labelledby="id..."` to give your `Modal` component an accessible name.
```

Reply:
Is `id` specifically required in the Modal's accessible name? Or is `id...` an agreed-upon placeholder for any name that the developer wants to use? _Is_ there a standard placeholder used in the MUI docs?

If not, I would suggest making it as obvious as possible that the dev can use whatever they want. Something like `aria-labelledby="MODAL_NAME_HERE"`.

Reply:
The id is the HTML id of the element describing it. It can be any arbitrary string.